### PR TITLE
Fixed AD users import

### PR DESCRIPTION
**Pull request by mpozhidaeva**
Fixed https://github.com/OHDSI/WebAPI/issues/598

### CODE REVIEW

File: src/main/java/org/ohdsi/webapi/user/importer/DefaultUserImporter.java [hunk 1]
```diff
@@ -81,16 +82,35 @@ public LdapTemplate getLdapTemplate(LdapProviderType providerType) {
...
-    AndFilter filter = new AndFilter();
-    filter.and(getCriteria(OBJECTCLASS_ATTR, getProvider(providerType).orElseThrow(IllegalArgumentException::new).getUserClass()));
-    CollectingNameClassPairCallbackHandler<LdapUser> handler =  provider.getUserSearchCallbackHandler(getUserAttributesMapper(providerType));
-    ldapTemplate.search(LdapUtils.emptyLdapName(), filter.encode(), provider.getUserSearchControls(), handler, new LdapTemplate.NullDirContextProcessor());
+    long countLimit = provider.getUserSearchControls().getCountLimit();
+    CollectingNameClassPairCallbackHandler<LdapUser> handler = provider.getUserSearchCallbackHandler(getUserAttributesMapper(providerType));
+    int resultsPerPage = 500;
+    PagedResultsDirContextProcessor pager = new PagedResultsDirContextProcessor(resultsPerPage);
+    do {
+      provider.search(getUserFilter(provider), handler, pager);
```
Comment (wivern) [inline src/main/java/org/ohdsi/webapi/user/importer/DefaultUserImporter.java]: why search moved to provider? this code contains no provider specific logic

File: src/main/java/org/ohdsi/webapi/user/importer/DefaultUserImporter.java [hunk 2]
```diff
@@ -81,16 +82,35 @@ public LdapTemplate getLdapTemplate(LdapProviderType providerType) {
...
+      pager = new PagedResultsDirContextProcessor(resultsPerPage, pager.getCookie());
+
+    } while (pager.getCookie() != null && pager.getCookie().getCookie() != null
+            && (countLimit == 0 || handler.getList().size() < countLimit));
+
     List<LdapUser> users = handler.getList();
+    if (countLimit > 0 && countLimit < users.size()) {
```
Comment (wivern) [inline src/main/java/org/ohdsi/webapi/user/importer/DefaultUserImporter.java]: finally it seems that ContextProcessor doesn't collect search results as expected,
pls check this suspicion

File: src/main/java/org/ohdsi/webapi/user/importer/DefaultUserImporter.java (hunk 1 above)
Comment (mpozhidaeva) [reply src/main/java/org/ohdsi/webapi/user/importer/DefaultUserImporter.java]: actually, the difference is in pager

File: src/main/java/org/ohdsi/webapi/user/importer/DefaultUserImporter.java [hunk 3]
```diff
@@ -81,16 +82,35 @@ public LdapTemplate getLdapTemplate(LdapProviderType providerType) {
...
+    long countLimit = provider.getUserSearchControls().getCountLimit();
+    CollectingNameClassPairCallbackHandler<LdapUser> handler = provider.getUserSearchCallbackHandler(getUserAttributesMapper(providerType));
+    int resultsPerPage = 500;
+    PagedResultsDirContextProcessor pager = new PagedResultsDirContextProcessor(resultsPerPage);
+    do {
+      provider.search(getUserFilter(provider), handler, pager);
+      pager = new PagedResultsDirContextProcessor(resultsPerPage, pager.getCookie());
+
+    } while (pager.getCookie() != null && pager.getCookie().getCookie() != null
+            && (countLimit == 0 || handler.getList().size() < countLimit));
```
[the remark this answers is not in the document]
Comment (mpozhidaeva) [reply src/main/java/org/ohdsi/webapi/user/importer/DefaultUserImporter.java]: hmm, don't see such method

**Pull request merged**